### DISCUSSION TITLE
Umbraco v9 Projects on .NET Standard 2.1 or .NET5

### DISCUSSION THREAD

**Discussion by nzdev** [umbraco/Umbraco-CMS#11099] [Features and ideas]
Not all of the Umbraco 9 packages are targeting a minimum of .NET Standard 2.1 or .NET 5. This will restrict access to API's going forward without causing a breaking change. This should be addressed before V9 is released. This will limit options to improve performance, especially with Span<T> code. Best as I can tell, the blocker is the support of SQL Compact which finished end of extended support on July 13 2021. https://docs.microsoft.com/en-us/lifecycle/products/microsoft-sql-server-compact-40 
Removing SQL Compact would be preferable and the least work over refactoring so that most of the code is targeting .net standard 2.1+ 

NET Core 2.1 added many APIs that allow working with spans, such as Stream.Read(Span<Byte>). Part of the value proposition to add span to .NET Standard is to add theses companion APIs as well.

https://devblogs.microsoft.com/dotnet/announcing-net-standard-2-1/

**Comment by nul800sebastiaan** [comment]
It's an unfortunate trade-off we have to make. We will still be supporting SQL CE to make setting up Umbraco (on Windows) super easy. We have database reworks in the planning and we'll definitely have to revisit what kind of DBs we support but we won't be dropping SQL CE in the near future I'm afraid.

**Comment by nzdev** [comment]
Found that it was possible to keep SQL CE and ensure .NET 5 targeting https://github.com/umbraco/Umbraco-CMS/pull/11128
